staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -191,6 +226,9 @@ func (r *withRetry) prepareForNextRetry(ctx context.Context, request *Request) e
 
 func (r *withRetry) Before(ctx context.Context, request *Request) error {
```

Comment:
Do we ever want `Before` to return something other than the `WrapPreviousError`?

Reply:
the call site `Do` and `Watch` invoke `WrapPreviousError` if `Before` returns an error.

```
if err := r.retry.Before(ctx, r); err != nil {
			return nil, err
			return nil, r.retry.WrapPreviousError(err)
		}
```

I have changed `Before`,  it tracks the previous error now.

Reply:
> apparently streams don't retry on error:

We can continue the discussion here: https://github.com/kubernetes/kubernetes/issues/109021